shared/actions/chat.js

```diff
@@ -1043,7 +1042,7 @@ function * _ensureValidSelectedChat (onlyIfNoSelection: boolean) {
     }
 
     const firstGood = inbox.find(i => !i.get('isEmpty') || alwaysShow.has(i.get('conversationIDKey')))
-    if (firstGood) {
+    if (firstGood && !isMobile) {
```

Comment:
No, not quite.  Every time the inbox is updated, a convo is muted, or a convo is blocked, we end up in _ensureValidSelectedChat.  And of those three cases, we'd only want to deselect in the last one.  The logic inside _ensureValidSelectedChat checks whether we're in that case (the selected conversation has now been deleted), so that's why its logic needs to run all the way through.